packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts

```diff
@@ -387,6 +387,19 @@ export type ConnectButton_connectModalOptions = {
    */
   privacyPolicyUrl?: string;
 
+  /**
+   * Require terms of service and privacy policy to be accepted before connecting an in-app wallet.
+   *
+   * By default it's `false`
+   * @example
+   * ```tsx
+   * <ConnectButton connectModal={{
+   *  requireApproval: true
+   * }} />
+   * ```
+   */
+  requireApproval?: boolean;
```

Comment:
If this only affects in-app wallet - shouldn't this option be in `inAppWallet()` ?

Reply:
seems to me more related to the TOS / policy links you can add no? dont feel like its inapp specific

Reply:
Yeah we should minimize UI stuff on the inAppWallet constructor

Reply:
In that case, we should disable connecting all the wallets if TOS is not checked right - and not just inApp wallet?

Reply:
Also - noticed that the passkey option is not disabled